File: 2029-stone-game-ix/2029-stone-game-ix.js
/**
 * @param {number[]} stones
 * @return {boolean}
 */
var stoneGameIX = function(stones) {
    const cnt = [0, 0, 0];

    for (const x of stones) {
        cnt[x % 3]++;
    }

    if (Math.min(cnt[1], cnt[2]) === 0) {
        return Math.max(cnt[1], cnt[2]) > 2 && cnt[0] % 2 === 1;
    }

    return Math.abs(cnt[1] - cnt[2]) > 2 || cnt[0] % 2 === 0;
};

File: 2029-stone-game-ix/2029-stone-game-ix.c
#include <stdbool.h>
#include <stdlib.h>

bool stoneGameIX(int* stones, int stonesSize) {
    int cnt[3] = {0, 0, 0};

    for (int i = 0; i < stonesSize; i++) {
        cnt[stones[i] % 3]++;
    }

    int min12 = cnt[1] < cnt[2] ? cnt[1] : cnt[2];
    int max12 = cnt[1] > cnt[2] ? cnt[1] : cnt[2];

    if (min12 == 0) {
        return max12 > 2 && cnt[0] % 2 == 1;
    }

    return abs(cnt[1] - cnt[2]) > 2 || cnt[0] % 2 == 0;
}